### DISCUSSION TITLE
How to split window

### DISCUSSION THREAD

**Discussion by sasakiyori** [bevyengine/bevy#8754] [Q&A] [answered]
I want to display score or something on the window. Sprites should not go to the position where score text displayed otherwise the overlap will occur. 

Is there a method to split the window just like below? Or should I just create a new window?

![image](https://github.com/bevyengine/bevy/assets/22006360/19fe4364-fa99-4992-80ad-9df96afb7fca)

**Comment by rlidwka** [comment, marked as answer]
You need two cameras with two different viewports.

Here is an example how to do that:
https://github.com/bevyengine/bevy/blob/main/examples/3d/split_screen.rs

**Reply by sasakiyori** [reply]
I found this example split screen by changing the viewport of the cameras. But it seems not split the primary window, which is a little different from what I expect.

If I use two cameras to do this, I should manually calculate the virtual boundary sprites can't go?  

Actually I hope I can get a specific query for window, whose size is the size of split window, so that sprites can easily move inside it. Or I can easily get the size of each window/viewport.

```rust
// implements like this:
fn sprite_move(
    mut sprite_transform: Query<&mut Transform, With<SomeSprite>>,
    window_query: Query<&Window, With<SplitWindow>>,
) {
}
```

I am new to bevy and lack understanding to it. Thanks in advance!

**Reply by rlidwka** [reply]
> If I use two cameras to do this, I should manually calculate the virtual boundary sprites can't go?

If you want game objects to be contained inside screen at all times, instead of Window size, you'd use Viewport physical_size.

> window_query: Query<&Window, With<SplitWindow>>,

There are no "split windows" in bevy as far as I know of. There are cameras, which you can restrict to specific rectangle viewport within your single window.

So it would be more like this:

```rs
fn sprite_move(
  mut sprite_transform: Query<&mut Transform, With<SomeSprite>>,
  camera_query: Query<&Camera, With<BottomCamera>>,
) {
    for camera in camera_query.iter() {
        dbg!(camera.viewport.physical_size);
    }
}
```

**Comment by reynadev-oss** [comment]
So in Bevy, rather than creating multiple windows, you can use a single window with two cameras, each assigned a different viewport. To manage where [sprites ](https://smartwindows.app/blog/how-to-split-screen-in-windows-10-and-11/)can move within each split screen, you can access the physical_size of the camera’s viewport (via camera.viewport.physical_size) and use that to constrain sprite movement—much cleaner than trying to manage separate windows or guessing boundaries.
